src/app/machines/CreateConnectorWizard.machine.ts

```diff
@@ -55,25 +56,37 @@ type Context = {
   duplicateMode?: boolean;
 };
 
+type JumpEvent = {
+  fromStep?: string;
+};
+
+type AnalyticsEvent = {
+  analyticsEventName: string;
+  [key: string]: unknown;
+};
+
 const model = createModel({} as Context, {
   events: {
     isValid: () => ({}),
     isInvalid: () => ({}),
     prev: () => ({}),
     next: () => ({}),
     changedStep: ({ step }: { step: number }) => ({ step }),
-    jumpToSelectKafka: () => ({}),
-    jumpToSelectNamespace: () => ({}),
-    jumpToSelectConnector: () => ({}),
-    jumpToConfigureConnector: ({ subStep }: { subStep?: number }) => ({
-      subStep,
-    }),
-    jumpToBasicConfiguration: () => ({}),
-    jumpToErrorConfiguration: () => ({}),
-    jumpToReviewConfiguration: () => ({}),
+    jumpToSelectKafka: () => ({} as JumpEvent),
+    jumpToSelectNamespace: () => ({} as JumpEvent),
+    jumpToSelectConnector: () => ({} as JumpEvent),
+    jumpToConfigureConnector: ({ subStep }: { subStep?: number }) =>
+      ({
+        subStep,
+      } as JumpEvent & { subStep?: number }),
+    jumpToCoreConfiguration: () => ({} as JumpEvent),
+    jumpToErrorConfiguration: () => ({} as JumpEvent),
+    jumpToReviewConfiguration: () => ({} as JumpEvent),
+    sendAnalytics: () => ({} as AnalyticsEvent),
   },
   actions: {
     notifySave: () => ({}),
+    sendAnalytics: () => ({}),
```

Comment:
@riccardo-forina when you're back around, do you think using a service instead of this action/event combo is a better approach?

Reply:
Maybe, yeah. `sendParent` is neat but leads to coupling. In the next version of XState there will be some kind of support for this kind of "systems of machines", but right now a service that will have to be injected by the parent machine sounds like something to try.

Reply:
Cool, thanks!  Yeah, think I'll rework that, as I agree that coupling you mention has been nagging at me :smile: